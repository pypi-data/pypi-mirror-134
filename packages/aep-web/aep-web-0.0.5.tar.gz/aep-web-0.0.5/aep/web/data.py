from functools import lru_cache
from glob import glob
from pathlib import Path
from typing import Dict, List, Text, Tuple

import aep.tools.libs.data


def get_bundles(args) -> List[Text]:
    bundle_list = []
    for bundle_type in ("incident", "threatactors"):
        for bundle in glob(f"{args.data_dir}/{bundle_type}/*"):
            bundle_list.append(Path(bundle_type) / Path(bundle).name)

    return sorted(bundle_list)


@lru_cache()
def read_promise_description_file(filename: Path) -> List:
    return aep.tools.libs.data.read_promise_description_file(open(Path(filename)))


@lru_cache()
def read_technique_promises(
    tech_promises_file: Path, promise_descriptions_file: Path, conditions_file: Path
) -> Tuple[Dict, Dict, bool]:

    return aep.tools.libs.data.read_technique_promises(
        tech_promises_file, promise_descriptions_file, conditions_file
    )
