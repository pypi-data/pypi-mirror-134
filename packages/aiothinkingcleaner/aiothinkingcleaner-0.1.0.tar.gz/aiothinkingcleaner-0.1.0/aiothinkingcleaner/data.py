from enum import Enum


class TCCommand(Enum):
    """Enum of commands available."""

    CLEAN = "clean"
    MAX = "max"
    DELAYED_CLEAN = "delayedclean"
    SPOT = "spot"
    DOCK = "dock"
    FIND_ME = "find_me"
    STOP = "stop"
    EXIT_DOCK = "leavehomebase"
    POWER_OFF = "poweroff"


class TCDeviceState(Enum):
    """Current device state of the Roomba."""

    BASE = "st_base"
    BASE_RECON = "st_base_recon"
    BASE_FULL = "st_base_full"
    BASE_TRICKLE = "st_base_trickle"
    BASE_WAIT = "st_base_wait"
    PLUG = "st_plug"
    PLUG_RECON = "st_plug_recon"
    PLUG_FULL = "st_plug_full"
    PLUG_TRICKLE = "st_plug_trickle"
    PLUG_WAIT = "st_plug_wait"
    STOPPED = "st_stopped"
    CLEANING = "st_clean"
    SPOT_CLEANING = "st_clean_spot"
    MAX_CLEANING = "st_clean_max"
    DOCKING = "st_dock"
    PICK_UP = "st_pickup"
    REMOTE = "st_remote"
    WAIT = "st_wait"
    OFF = "st_off"
    CLEAN_STOP = "st_cleanstop"
    LOCATE = "st_locate"
    UNKNOWN = "st_unknown"
    ERROR_CHARGE = "error_charge"
    ERROR_LEFT_L = "error_left_l"
    ERROR_RIGHT_L = "error_right_l"
    ERROR_CLIFF = "error_cliff"
    ERROR_LEFT_W = "error_left_w"
    ERROR_RIGHT_W = "error_right_w"
    ERROR_BRUSH = "error_brush"
    ERROR_SIDE = "error_side"
    ERROR = "error"
    OFFLINE = "offline"


class TCDeviceStatus:
    """Current status of the Roomba."""

    name: str
    """The name of the device."""

    battery_charge: float
    """The battery percentage (between 0 and 100)."""

    capacity: int
    """The current battery capacity in mAh."""

    state: TCDeviceState
    """The current state of the Roomba."""

    is_cleaning: bool
    """True when the Roomba is currently cleaning."""

    is_near_homebase: bool
    """True when the Roomba has visual contact with a homebase."""

    def __init__(
        self,
        name: str,
        battery_charge: float,
        capacity: int,
        cleaner_state: str,
        cleaning: str,
        near_homebase: str,
    ) -> None:
        self.name = name
        self.battery_charge = battery_charge
        self.capacity = capacity
        self.state = TCDeviceState(cleaner_state)
        self.is_cleaning = cleaning == "1"
        self.is_near_homebase = near_homebase == "1"
