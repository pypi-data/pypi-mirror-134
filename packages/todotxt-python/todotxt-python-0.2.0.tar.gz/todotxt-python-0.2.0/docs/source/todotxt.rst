todotxt package
===============

.. automodule:: todotxt
   :members:
   :undoc-members:
   :show-inheritance:
