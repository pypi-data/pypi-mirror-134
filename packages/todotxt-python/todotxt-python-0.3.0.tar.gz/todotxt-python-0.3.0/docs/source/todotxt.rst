todotxt package
===============

.. automodule:: todotxt
   :members:
   :undoc-members:
   :show-inheritance:

Submodules
----------

.. toctree::
   :maxdepth: 4

   todotxt.types
