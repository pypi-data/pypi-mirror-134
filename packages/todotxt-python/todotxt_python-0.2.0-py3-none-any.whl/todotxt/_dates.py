"""Datetime utilities used throughout this library."""

from __future__ import annotations

import datetime as dt
from typing import Final


DATE_FMT: Final = "%Y-%m-%d"
RE_DATE: Final = "[1-9][0-9][0-9][0-9]-[0-1][0-9]-[0-3][0-9]"


def to_date(YYYYMMDD: str) -> dt.date:
    """Helper function for constructing a date object."""
    return dt.datetime.strptime(YYYYMMDD, DATE_FMT).date()
