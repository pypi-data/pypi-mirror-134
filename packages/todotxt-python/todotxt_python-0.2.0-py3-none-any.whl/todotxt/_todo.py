"""Contains the Todo class definition."""

from __future__ import annotations

from dataclasses import dataclass
import datetime as dt
import re
from typing import Dict, Final, List, Optional, Tuple, cast

from eris import ErisError, Err, Ok, Result

from ._dates import RE_DATE, to_date
from .types import Priority


DEFAULT_PRIORITY: Final[Priority] = "N"
PROJECT_PREFIX: Final = "+"
CONTEXT_PREFIX: Final = "@"

RE_TODO: Final = r"""
(?P<x>x[ ]+)?                        # optional 'x'
\((?P<priority>[A-Z])\)[ ]+          # priority
(?:
    (?:(?P<done_date>{0})[ ]+)?      # optional date of completion
    (?:(?P<create_date>{0})[ ]+)     # optional date of creation
)?
(?P<desc>.*)                         # description
""".format(
    RE_DATE
)


@dataclass(frozen=True)
class Todo:
    """Represents a single task in a todo list."""

    desc: str

    contexts: Tuple[str, ...] = ()
    create_date: Optional[dt.date] = None
    done_date: Optional[dt.date] = None
    marked_done: bool = False
    metadata: Optional[Dict[str, str]] = None
    priority: Priority = DEFAULT_PRIORITY
    projects: Tuple[str, ...] = ()

    @classmethod
    def from_string(cls, string: str) -> Result[Todo, ErisError]:
        """Contructs a Todo object from a string (usually a line in a file)."""
        re_todo_match = re.match(RE_TODO, string, re.VERBOSE)
        if re_todo_match is None:
            return Err(
                f"The provided string ({string!r}) does not appear to properly"
                " adhere to the todo.txt format. See"
                " https://github.com/todotxt/todo.txt for the specification."
            )

        marked_done: bool = False
        if re_todo_match.group("x"):
            marked_done = True

        priority: Priority = DEFAULT_PRIORITY
        if grp := re_todo_match.group("priority"):
            priority = cast(Priority, grp)

        create_date: Optional[dt.date] = None
        if grp := re_todo_match.group("create_date"):
            create_date = to_date(grp)

        done_date: Optional[dt.date] = None
        if grp := re_todo_match.group("done_date"):
            done_date = to_date(grp)

        desc = re_todo_match.group("desc")
        all_words = desc.split(" ")

        project_list: List[str] = []
        context_list: List[str] = []
        for some_list, prefix in [
            (project_list, PROJECT_PREFIX),
            (context_list, CONTEXT_PREFIX),
        ]:
            for word in all_words:
                if word.startswith(prefix):
                    some_list.append(word[len(prefix) :])

        projects = tuple(project_list)
        contexts = tuple(context_list)

        metadata: Optional[Dict[str, str]] = None
        mdata: Dict[str, str] = {}
        for word in all_words:
            kv = word.split(":", maxsplit=1)
            if len(kv) == 2:
                key, value = kv
                mdata[key] = value

        if mdata:
            metadata = mdata

        todo = Todo(
            contexts=contexts,
            create_date=create_date,
            desc=desc,
            done_date=done_date,
            marked_done=marked_done,
            metadata=metadata,
            priority=priority,
            projects=projects,
        )
        return Ok(todo)
