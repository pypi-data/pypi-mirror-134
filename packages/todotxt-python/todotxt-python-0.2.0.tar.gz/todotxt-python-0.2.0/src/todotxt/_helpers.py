"""Helper utilities provided by the todotxt library."""

from pathlib import Path
from typing import List

from eris import Err
from typist import PathLike

from ._todo import Todo


def read_todos_from_file(path_like: PathLike) -> List[Todo]:
    """Reads all todo lines from a given file.

    Pre-conditions:
        * `path_like` exists and is a file.
    """
    path = Path(path_like)

    assert (
        path.is_file()
    ), f"The provided path does not exist or is not a file: {path}"

    todos_in_file = []
    for line in path.read_text().split("\n"):
        line = line.strip()
        todo_result = Todo.from_string(line)
        if isinstance(todo_result, Err):
            continue

        todo = todo_result.ok()
        todos_in_file.append(todo)

    return todos_in_file
