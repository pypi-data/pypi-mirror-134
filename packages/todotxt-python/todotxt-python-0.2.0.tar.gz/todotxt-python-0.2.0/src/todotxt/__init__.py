"""A Python library for working with the todo.txt format."""

import logging as _logging

from . import types
from ._helpers import read_todos_from_file
from ._todo import DEFAULT_PRIORITY, Todo


__all__ = ["DEFAULT_PRIORITY", "Todo", "read_todos_from_file", "types"]

__author__ = "Bryan M Bugyi"
__email__ = "bryanbugyi34@gmail.com"
__version__ = "0.2.0"

_logging.getLogger(__name__).addHandler(_logging.NullHandler())
