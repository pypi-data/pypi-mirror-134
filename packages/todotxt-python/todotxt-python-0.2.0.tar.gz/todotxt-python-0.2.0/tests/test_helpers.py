"""Test the helper utilities provided by the todotxt library."""

from __future__ import annotations

from pathlib import Path

from pytest import mark
from syrupy.assertion import SnapshotAssertion as Snapshot

import todotxt

from .data import get_all_todo_files


params = mark.parametrize


@params("todo_file", get_all_todo_files())
def test_read_todos_from_file(snapshot: Snapshot, todo_file: Path) -> None:
    """Test the read_todos_from_file() function."""
    todos = todotxt.read_todos_from_file(todo_file)
    assert [todo.__dict__ for todo in todos] == snapshot
