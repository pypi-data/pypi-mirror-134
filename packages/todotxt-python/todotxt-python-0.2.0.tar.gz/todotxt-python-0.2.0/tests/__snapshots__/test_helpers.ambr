# name: test_read_todos_from_file[todo_file0]
  <class 'list'> [
    <class 'dict'> {
      'contexts': <class 'tuple'> (
      ),
      'create_date': None,
      'desc': 'First item.',
      'done_date': None,
      'marked_done': False,
      'metadata': None,
      'priority': 'N',
      'projects': <class 'tuple'> (
      ),
    },
    <class 'dict'> {
      'contexts': <class 'tuple'> (
        'high',
      ),
      'create_date': None,
      'desc': 'High-priority item @high',
      'done_date': None,
      'marked_done': False,
      'metadata': None,
      'priority': 'A',
      'projects': <class 'tuple'> (
      ),
    },
    <class 'dict'> {
      'contexts': <class 'tuple'> (
      ),
      'create_date': datetime.date(2022, 1, 1),
      'desc': 'Done item',
      'done_date': datetime.date(2022, 1, 10),
      'marked_done': True,
      'metadata': None,
      'priority': 'B',
      'projects': <class 'tuple'> (
      ),
    },
    <class 'dict'> {
      'contexts': <class 'tuple'> (
      ),
      'create_date': datetime.date(2022, 1, 11),
      'desc': 'Item is due:2022-01-31',
      'done_date': None,
      'marked_done': False,
      'metadata': <class 'dict'> {
        'due': '2022-01-31',
      },
      'priority': 'N',
      'projects': <class 'tuple'> (
      ),
    },
    <class 'dict'> {
      'contexts': <class 'tuple'> (
        'stuff',
      ),
      'create_date': datetime.date(1900, 1, 1),
      'desc': 'This +test item is awesome:True and has lots of @stuff too_long:NoWay DoFoo:22',
      'done_date': datetime.date(1999, 1, 1),
      'marked_done': True,
      'metadata': <class 'dict'> {
        'DoFoo': '22',
        'awesome': 'True',
        'too_long': 'NoWay',
      },
      'priority': 'Z',
      'projects': <class 'tuple'> (
        'test',
      ),
    },
  ]
---
